crates/bins/node/src/rest_service.rs

```diff
@@ -0,0 +1,504 @@
+#[cfg(feature = "rest")]
+use axum::{
+    Router,
+    extract::{Path, State},
+    http::StatusCode,
+    response::Json,
+    routing::{get, post},
+};
+
+#[cfg(feature = "rest")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "rest")]
+use nuts::{
+    Amount,
+    nut00::{BlindedMessage, Proof, secret::Secret},
+    nut01::PublicKey,
+    nut02::KeysetId,
+    nut19::Route,
+};
+
+#[cfg(feature = "rest")]
+use std::str::FromStr;
+
+#[cfg(feature = "rest")]
+use uuid::Uuid;
+
+#[cfg(feature = "rest")]
+use crate::{app_state::AppState, response_cache::CachedResponse};
+
+#[cfg(feature = "rest")]
+use node::{
+    BlindSignature as GrpcBlindSignature, KeysetKeys as GrpcKeysetKeys,
+    SwapResponse as GrpcSwapResponse,
+};
+
+#[cfg(feature = "rest")]
+use starknet_types::Unit;
+
+/// HTTP error response
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct ErrorResponse {
+    error: String,
+}
+
+/// Cashu NUT-01: Get keysets
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct GetKeysetsResponse {
+    keysets: Vec<KeysetInfo>,
+}
+
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct KeysetInfo {
+    id: String,
+    unit: String,
+    active: bool,
+}
+
+/// Cashu NUT-01: Get keys for keysets
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct GetKeysResponse {
+    keysets: Vec<KeysetKeys>,
+}
+
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct KeysetKeys {
+    id: String,
+    unit: String,
+    active: bool,
+    keys: std::collections::HashMap<u64, String>,
+}
+
+/// Cashu NUT-03: Swap request
+#[cfg(feature = "rest")]
+#[derive(Deserialize, Debug)]
+struct SwapRequest {
+    inputs: Vec<ProofHttp>,
+    outputs: Vec<BlindedMessageHttp>,
+}
+
+/// Cashu NUT-03: Swap response
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct SwapResponse {
+    signatures: Vec<BlindSignatureHttp>,
+}
+
+/// HTTP representation of Proof
+#[cfg(feature = "rest")]
+#[derive(Deserialize, Debug)]
+struct ProofHttp {
+    amount: u64,
+    #[serde(rename = "id")]
+    keyset_id: String,
+    secret: String,
+    #[serde(rename = "C")]
+    c: String,
+}
+
+/// HTTP representation of BlindedMessage
+#[cfg(feature = "rest")]
+#[derive(Deserialize, Debug)]
+struct BlindedMessageHttp {
+    amount: u64,
+    #[serde(rename = "id")]
+    keyset_id: String,
+    #[serde(rename = "B_")]
+    blinded_secret: String,
+}
+
+/// HTTP representation of BlindSignature
+#[cfg(feature = "rest")]
+#[derive(Serialize)]
+struct BlindSignatureHttp {
+    amount: u64,
+    #[serde(rename = "id")]
+    keyset_id: String,
+    #[serde(rename = "C_")]
+    c: String,
+}
```

Comment:
Why can't you just use the ones present in the `nuts` crate?
I think it would be better if possilbe